subprojects/core/src/test/groovy/org/gradle/api/tasks/AbstractCopyTaskTest.groovy

```diff
@@ -33,22 +36,28 @@ class AbstractCopyTaskTest extends WorkspaceTest {
         hasDeclaredOutputs() >> true
     }
     TestCopyTask task
+    TestFile projectDir
 
     def setup() {
-        task = TestUtil.create(temporaryFolder).task(TestCopyTask)
+        projectDir = file("project").createDir()
```

Comment:
❌ What was the problem here? Shouldn't this test be annotated with `@CleanupTestDirectory` and
`@UsesNativeServices` as `AbstractProjectBuilderSpec` is?

Reply:
The problem was that the `copy spec methods delegate to main spec of copy action` test failed because `testDirectory` contained `user-home` which now contains a `caches/journal-1/meta.properties`. I don't think `@CleanupTestDirectory` or `@UsesNativeServices` would help, would they?

Reply:
Makes sense.

If you use project builder you also should annotate the class with `@UsesNativeServices`, so these are correctly initialized. `@CleanupTestDirectory` is nice since it prevents cleaning up the test directory when the test failed - so the entries in the test directory can be examined.

I suggest adding both to the test.

Reply:
Done in 414f617b2357fab893bc3360b113a5cea3f2f4b8.